### PR TITLE
US - 0.7.0 - Events and event details

### PR DESCRIPTION
**Pull request by med-amine-bj**
US: https://trello.com/c/2jmArMzJ/28-us-070-news

### CODE REVIEW

File: Usiqee/Manager/ManagerNews.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+//  Created by Amine on 29/06/2021.
+//
+
+import Foundation
+
+protocol ManagerNewsDelegate: AnyObject {
+    func didUpdateNews()
+    func didStartLoading()
+    func didFinishLoading()
+    func didStartLoadMoreLoading()
```
Comment (guaire94) [inline Usiqee/Manager/ManagerNews.swift:14]: `didLoadMore` instead of `didStartLoadMoreLoading`?

File: Usiqee/Model/Author/Author.swift [hunk 2]
```diff
@@ -8,13 +8,99 @@
...
+                return UIImage(named: "snapchat")
+            case .twitch:
+                return UIImage(named: "twitch")
+            case .tiktok:
+                return UIImage(named: "tiktok")
+            case .linkedin:
+                return UIImage(named: "linkedin")
+            }
+        }
+    }
```
Comment (guaire94) [inline Usiqee/Model/Author/Author.swift:43]: should be rename `AuthorSocialMedia` or put in a new file `SocialMedia`

File: Usiqee/Model/Author/Author.swift [hunk 3]
```diff
@@ -8,13 +8,99 @@
...
+    let linkedinLink: String?
+    let webLink: String?
+    
+    // MARK: - Helper
+    var hasExternalLinks: Bool {
+        if twitterLink == nil && fbLink == nil && instagramLink == nil && youtubeLink == nil && snapchatLink == nil && twitchLink == nil && tiktokLink == nil && linkedinLink == nil {
+            return false
+        }
+        
+        return true
```
Comment (guaire94) [inline Usiqee/Model/Author/Author.swift:66]: less optimize but better readbility
```suggestion
       !externalLinks.isEmpty
```

File: Usiqee/Module/News/NewsDetails/Cells/Author/NewsDetailsAuthorCell.swift [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    
+    private func hideSubviews() {
+        externalLinkView.isHidden = true
+        authorLinksView.isHidden = true
+        authorLinksStack.arrangedSubviews.forEach { $0.removeFromSuperview() }
+    }
+    
+    private func showExternalLinks(author: Author) {
+        author.externalLinks.forEach { externalLink in
+            let button = UIButton(frame: CGRect(x: 0, y: 0, width: 35, height: 35))
```
Comment (guaire94) [inline Usiqee/Module/News/NewsDetails/Cells/Author/NewsDetailsAuthorCell.swift]: can we create  `static let externalLinkButtonFrame = CGRect(x: 0, y: 0, width: 35, height: 35)` in Constants ?

File: Usiqee/Model/Author/Author.swift [hunk 5]
```diff
@@ -8,13 +8,99 @@
...
+    }
+    
+    var externalLinks: [SocialMedia] {
```
Comment (guaire94) [inline Usiqee/Model/Author/Author.swift:69]: can we create a tupple <SocialMedia - Url> or store link when adding SocialMedia
Will be more easy to use in this example: (NewsDetailsAuthorCell)
 
```
let url: String?
        switch item {
        case .twitter:
            url = author?.twitterLink
        case .facebook:
            url = author?.fbLink
        case .instagram:
            url = author?.instagramLink
        case .youtube:
            url = author?.youtubeLink
        case .snapchat:
            url = author?.snapchatLink
        case .twitch:
            url = author?.twitchLink
        case .tiktok:
            url = author?.tiktokLink
        case .linkedin:
            url = author?.linkedinLink
        }
```

File: Usiqee/Module/News/NewsDetails/NewsDetailsTableViewHandler.swift [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        case author(Author)
+    }
+    
+    // MARK: - Properties
+    var news: NewsItem?
+    var sections: [NewsSection] = []
+    var author: Author?
+    
+    // MARK: - Helper
+    func numberOfRows() -> Int {
```
Comment (guaire94) [inline Usiqee/Module/News/NewsDetails/NewsDetailsTableViewHandler.swift:29]: can be computed var

File: Usiqee/Service/ServiceNews.swift [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        var listener = FFirestoreReference.news
+            .whereField("isDraft", isEqualTo: false)
+            .order(by: "date", descending: true)
+        
+        if let lastSnapshot = lastSnapshot {
+            listener = listener.start(afterDocument: lastSnapshot)
+        }
+        
+        self.listener = listener
+            .limit(to: 10)
```
Comment (guaire94) [inline Usiqee/Service/ServiceNews.swift:38]: can be put in constant

File: Usiqee/Model/Author/Author.swift (hunk 5 above)
Comment (med-amine-bj) [reply Usiqee/Model/Author/Author.swift:68]: I add it to author

File: Usiqee/Model/Author/Author.swift (hunk 2 above)
Comment (med-amine-bj) [reply Usiqee/Model/Author/Author.swift]: it already inside author (should be prefixed by Author `Author.SocialMedia`)

**Review by guaire94 — APPROVED** [review]

**Pull request merged**